friendships.go

```diff
@@ -0,0 +1,241 @@
+	stringifyIDs bool) (*UserIDsPageResponse, error) {
+	return c.friendshipsDirection(ctx, "outgoing", cursor, stringifyIDs)
+}
+
+// FriendshipsCreate calls the Twitter /friendships/create.json endpoint.
+func (c *Client) FriendshipsCreate(ctx context.Context, screenName,
+	userID string, follow bool) (*UserResponse, error) {
+	values := url.Values{}
+	if screenName != "" {
+		values.Set("screen_name", screenName)
+	}
+	if userID != "" {
+		values.Set("user_id", userID)
+	}
+	values.Set("follow", strconv.FormatBool(follow))
+	resp, err := c.do(ctx, "POST",
+		"https://api.twitter.com/1.1/friendships/create.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res User
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return nil, err
+	}
+	return &UserResponse{
+		User:      res,
+		RateLimit: getRateLimit(resp.Header),
+	}, nil
+}
+
+// FriendshipsDestroy calls the Twitter /friendships/destroy.json endpoint.
+func (c *Client) FriendshipsDestroy(ctx context.Context, screenName,
+	userID string) (*UserResponse, error) {
+	values := url.Values{}
+	if screenName != "" {
+		values.Set("screen_name", screenName)
+	}
+	if userID != "" {
+		values.Set("user_id", userID)
+	}
+	resp, err := c.do(ctx, "POST",
+		"https://api.twitter.com/1.1/friendships/destroy.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res User
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return nil, err
+	}
+	return &UserResponse{
+		User:      res,
+		RateLimit: getRateLimit(resp.Header),
+	}, nil
+}
+
+type friendshipsDetail struct {
+	RelationshipDetail `json:"relationship"`
+}
+
+// FriendshipsUpdate calls the Twitter /friendships/update.json endpoint.
+func (c *Client) FriendshipsUpdate(ctx context.Context, screenName,
+	userID string, device, retweets bool) (*RelationshipDetailResponse, error) {
+	values := url.Values{}
+	if screenName != "" {
+		values.Set("screen_name", screenName)
+	}
+	if userID != "" {
+		values.Set("user_id", userID)
+	}
+	values.Set("device", strconv.FormatBool(device))
+	values.Set("retweets", strconv.FormatBool(retweets))
+
+	resp, err := c.do(ctx, "POST",
+		"https://api.twitter.com/1.1/friendships/update.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res friendshipsDetail
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return nil, err
+	}
+	return &RelationshipDetailResponse{
+		Relationship: res.RelationshipDetail,
+		RateLimit:    getRateLimit(resp.Header),
+	}, nil
+}
+
+// FriendshipsShow calls the Twitter /friendships/show.json endpoint.
+func (c *Client) FriendshipsShow(ctx context.Context, sourceID, targetID int64,
```

Comment:
We've been using IDs as strings everywhere, so I'd recommend passing sourceID/targetID as strings here as well.

https://dev.twitter.com/overview/api/twitter-ids-json-and-snowflake